### PR TITLE
Add DisposeScopeOnAppDisposing to AutofacWebApiAppBuilderExtensions.

### PR DESCRIPTION
**Pull request by chriseldredge**
Implements #1.

### CODE REVIEW

File: src/Autofac.Integration.WebApi.Owin/AutofacWebApiAppBuilderExtensions.cs [hunk 1]
```diff
@@ -57,5 +60,28 @@ public static IAppBuilder UseAutofacWebApi(this IAppBuilder app, HttpConfigurati
...
+        /// convenience method that will dispose an Autofac container or child scope
+        /// when an OWIN application is shutting down.
+        /// </summary>
+        /// <param name="app">The application builder.</param>
+        /// <param name="lifetimeScope">The Autofac lifetime scope that should be disposed.</param>
+        /// <returns>The application builder.</returns>
+        [SecuritySafeCritical]
+        public static IAppBuilder DisposeScopeOnAppDisposing(this IAppBuilder app, ILifetimeScope lifetimeScope)
+        {
+            var context = new OwinContext(app.Properties);
```
Comment (tillig) [inline src/Autofac.Integration.WebApi.Owin/AutofacWebApiAppBuilderExtensions.cs:79]: Check `lifetimeScope` for null, throw `ArgumentNullException`

**Comment by tillig** [comment]
This looks awesome. If we can get a null check on `lifetimeScope` in there, I'll merge.

**Comment by chriseldredge** [comment]
Added checks for both arguments since it is technically possible (though unlikely) to invoke an extension method on a null instance of IAppBuilder.

**Comment by tillig** [comment]
_Rock and roooooooll_ :guitar:

**Comment by tillig** [comment]
I'm realizing as I read this that it probably should be in Autofac.Owin rather than in the Web API specific  library. My bad on not seeing that before.

When I get this library ready to go for 4.0.0 I may move the method so folks using OWIN but not necessarily Web API can make use of it.

**Pull request merged**